src/mapgen.cpp

```diff
@@ -4241,12 +4241,10 @@ ff.......|....|WWWWWWWW|\n\
                     // ---     -+-
                     // -+- ==> -+-
                     // ...     ...
-                    if( ( ter( doorx + i, doory ) == t_concrete_wall || ter( doorx + i, doory ) == t_concrete_wall ) &&
```

Comment:
Urgh, my mistake, there's an extra level of parentheses in there I missed. Never mind!